lcls-twincat-optics/lcls_twincat_optics_plc/POUs/FB_PitchControl.TcPOU

```diff
@@ -0,0 +1,237 @@
+			fbFormatString.arg1 := F_LREAL(lrCurrentSetpoint);
+			fbFormatString.sFormat := 'OoR, using stepper %f';
+			fbFormatString(sOut=>stDiag.asResults[stDiag.resultIdx.IncVal()]);
+			PC_State := PCM_Coarse50Piezo;
+		END_IF
+	PCM_Coarse50Piezo:
+		// A coarse move uses the stepper to do a best-effort position
+		// First set the piezo to nominal 50% extension using idle mode
+		//////////////////////////////////////////////////////////////////////////////
+		Pitch.Piezo.xIdleMode := TRUE;
+		// Indicate we are doing the coarse 50% piezo move	
+		bCoarse50PiezoMove := TRUE;
+		// Wait for piezo to settle
+		tonPiezoSettled.IN := TRUE;
+		bCoarse50PiezoMove R= tonPiezoSettled.Q;
+		IF tonPiezoSettled.Q THEN
+			//Piezo has moved to 50% position, finish with the stepper
+			PC_State := PCM_CoarseMove;
+			tonPiezoSettled.IN := FALSE;
+		END_IF
+	PCM_CoarseMove: 
+		// With the piezo at a nominal 50% extension, move the stepper to requested position
+		bExecuteStepper := TRUE;
+		// Timer that waits to start until stepper is within range of the setpoint
+		tonStepperHold.IN := WithinRange(ValA:=LREAL_TO_REAL(lrActPos), Center:=lrCurrentSetpoint, Range:=rSettledRange, Offset:=0);
+		tonStepperHold(); // call this here to reset Q just below on first cycle
+		// If the coarse move is complete, finish position correction with the piezo
+		IF tonStepperHold.Q  OR ftLimitSwitch.Q THEN
+			PC_State := PCM_CoarseMoveCleanup;
+			lrPrevStepperPos := lrActPos;
+		ELSIF ftSTO.Q THEN
+			PC_State := PCM_STOHit;
+			lrPrevStepperPos := lrActPos;
+		ELSIF Stepper.bError THEN
+			bExecuteStepper := FALSE;
+			PC_State := PCM_StepperError;
+			// Left out logging
+		END_IF
+	PCM_CoarseMoveCleanup:
+		bExecuteStepper := FALSE;
+		PC_State := PCM_FineMove;
+	PCM_FineMove:	
+		Pitch.Piezo.xIdleMode := FALSE;
+		fbPiezoControl.xExecute := TRUE;
+		IF bLimitHit THEN
+			Pitch.Piezo.rReqAbsPos := lrActPos;
+		ELSE
+			Pitch.Piezo.rReqAbsPos := lrCurrentSetpoint;
+		END_IF
+		rtPiezoMoveDone(CLK:=fbPiezoControl.xDone);
+		IF rtPiezoMoveDone.Q THEN
+			fbPiezoControl.xExecute := FALSE;
+			PC_State := PCM_Done;
+		END_IF
+	PCM_Done:
+		// Set the previously requested position here
+		lrLastSetpoint := lrCurrentSetpoint;
+		bLimitHit := FALSE;
+		// Indicate we're done
+		q_bDone	:= TRUE;
+		// Move back to standby
+		PC_State := PCM_Standby;
+	PCM_StepperError:
+		PC_State := PCM_Init;
+	PCM_PiezoError:
+		PC_State := PCM_Init;
+	PCM_OtherError:
+		PC_State := PCM_Init;
+END_CASE
+
+fbMotionRequest(stMotionStage:=Stepper,
+                bExecute:=bExecuteStepper,
+                bReset:=bResetStepper,
+                enumMotionRequest:=enumMotionRequest,
+                fPos:=lrCurrentSetpoint,
+                fVel:=Stepper.fVelocity,
+                fAcc:=Stepper.fAcceleration,
+                fDec:=Stepper.fDeceleration,
+                bError=>,
+                sErrorMessage=>,
+                bBusy=>,
+                bDone=>);
```

Comment:
Using FB_MotionRequest like this implies that you are not using the motor record, which is ok but confusing.

Reply:
e.g. motor record requests will be sent directly to `DUT_MotionStage` and not look at this request block

Reply:
Why does it imply that? I call `FB_MotionStage` directly below, though its true my way of requesting positions is by writing to `DUT_MotionStage.fPosition`,  And then what would the usage look like if I were to be using the motor record?

Reply:
The existence of FB_MotionRequest in the file at all implies that you are executing it sometimes, which means you are asking for moves inside the PLC program instead of through EPICS.

Reply:
Resolved in commit a093995